lib/sinon/match.js

```diff
@@ -221,12 +221,50 @@ match.hasOwn = createPropertyMatcher(function (actual, property) {
     return actual.hasOwnProperty(property);
 }, "hasOwn");
 
+match.array = match.typeOf("array");
+
+match.array.deepEquals = function (expectation) {
+    return match(function (actual) {
+        // Comparing lengths is the fastest way to spot a difference before iterating through every item
+        var sameLength = actual.length === expectation.length;
+        return sameLength && Array.prototype.every.call(actual, function (element, index) {
```

Comment:
Well, you could still cache the references to the functions to make it a bit less verbose. 

```
const proto = Array.prototype;
const every = proto.every;
```